pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,491 @@
+)
+
+const (
+	// Name pod name
+	Name string = "Name"
+	// Ready pod ready
+	Ready string = "Ready"
+	// Status pod status
+	Status string = "Status"
+	// Restart pod restart
+	Restart string = "Restart"
+	// Age pod age
+	Age string = "Age"
+)
+
+type listFlags struct {
+	util.CommonFlags
+	// Namespace pod namespace
+	Namespace string
+	// JobName represents the pod created under this vcjob,
+	// filtered by volcano.sh/job-name label
+	// the default value is empty, which means
+	// that all pods under vcjob will be obtained.
+	JobName string
+	// allNamespace represents getting all namespaces
+	allNamespace bool
+	// QueueName represents queue name
+	QueueName string
+}
+
+var listPodFlags = &listFlags{}
+
+// InitListFlags init list command flags.
+func InitListFlags(cmd *cobra.Command) {
+	util.InitFlags(cmd, &listPodFlags.CommonFlags)
+
+	cmd.Flags().StringVarP(&listPodFlags.QueueName, "queue", "q", "", "list pod with specified queue name")
+	cmd.Flags().StringVarP(&listPodFlags.JobName, "job", "j", "", "list pod with specified job name")
+	cmd.Flags().StringVarP(&listPodFlags.Namespace, "namespace", "n", "default", "the namespace of job")
+	cmd.Flags().BoolVarP(&listPodFlags.allNamespace, "all-namespaces", "", false, "list jobs in all namespaces")
+}
+
+// ListPods lists all pods details created by vcjob
+func ListPods(ctx context.Context) error {
+	config, err := util.BuildConfig(listPodFlags.Master, listPodFlags.Kubeconfig)
+	if err != nil {
+		return err
+	}
+	if listPodFlags.allNamespace {
+		listPodFlags.Namespace = ""
+	}
+
+	var pods corev1.PodList
+	listPodFunc := func(labelSelector labels.Selector) (*corev1.PodList, error) {
+		client := kubeclientset.NewForConfigOrDie(config)
+		// Construct the list options based on label and annotation selectors
+		opts := metav1.ListOptions{
+			LabelSelector: labelSelector.String(),
+		}
+		return client.CoreV1().Pods(listPodFlags.Namespace).List(ctx, opts)
+	}
+
+	// Filter pods based on annotationKey and annotationValue
+	filterFunc := func(pods *corev1.PodList, annotationKey, annotationValue string) *corev1.PodList {
+		filteredPods := &corev1.PodList{}
+		for _, pod := range pods.Items {
+			if value, exist := pod.Annotations[annotationKey]; exist {
+				if value == annotationValue {
+					filteredPods.Items = append(filteredPods.Items, pod)
+				}
+			}
+		}
+		return filteredPods
+	}
+
+	// if job name are specified, use job name to filter pods
+	// if the job is specified, it means that no matter whether the queue name is specified or not,
+	// we can just use the job name query. Because vcjob itself will specify a specific queue
+	if listPodFlags.JobName != "" {
+		labelSelector, err := createPodLabelSelectorByJobName(listPodFlags.JobName)
+		if err != nil {
+			return err
+		}
+		listVcjobPodsRes, err := listPodFunc(labelSelector)
+		if err != nil {
+			return err
+		}
+		pods.Items = append(pods.Items, listVcjobPodsRes.Items...)
+	} else if listPodFlags.QueueName != "" {
```

Comment:
Here is too complex and not readable, just list all pods and the filter by queue is another way, which is more clear.

Reply:
fixed